### DISCUSSION TITLE
Setting consumer-id in KafkaConnector?

### DISCUSSION THREAD

**Discussion by dcoracle** [helidon-io/helidon#6943] [Q&A]
Hi, is there a way to set the consumer-id using the KafkaConnector?

I do not see it in the docs:
https://helidon.io/docs/v2/#/mp/reactivemessaging/04_kafka

Nor do I see it in KafkaConfigBuilder:
https://helidon.io/docs/v2/apidocs/io.helidon.messaging.connectors.kafka/io/helidon/messaging/connectors/kafka/KafkaConfigBuilder.html

**Comment by danielkec** [comment]
No it is not, it was possible with old versions of Kafka client but it has been intentionally removed.
https://stackoverflow.com/a/63115387

**Comment by danielkec** [comment]
Maybe `client.id` is what you wanted?

**Comment by dcoracle** [comment]
That can work.  Probably best to say my use case.  I am trying to create a generic admin microservice that outputs the consumer groups for a particular topic (for observability).  We have 3 replicas in a replicaset for some other microservice associated with a topic with 3 partitions and I just want to be able to ensure I can associate it from the logs of a particular pod to the partition

Is there a way to get a handle to the KafkaClient object that Helidon MP Bootstraps with and output the client.id in the logs?

**Reply by dcoracle** [reply]
By the way, I tried to output the client.id from the admin microservice and all 3 replicas contain the same value which unfortunately doesn't help the situation.  Seems like whatever defaults MP is using results in the same value for the same replicaset.  

So the question is now similar to the original question.  Is there a way to set the client.id in the KafkaConnector?  Would be ideal to populate it with the pod.id

**Reply by danielkec** [reply]
I am afraid we don't have such API, but I would guess each connected consumer is actually logging client.id. It is important to say that each Helidon instance can have more than one client - multiple channels means multiple clients
